### PR TITLE
Fixing how goal number is stored in the file

### PR DESCRIPTION
**Pull request by dtracers**

### CODE REVIEW

File: api/stats/events.proto [hunk 1]
```diff
@@ -35,8 +35,15 @@ message Hit {
...
 
+    // What goal number this hit belongs to:
+    // it starts at 0 and counts up,  -1 means no more goals were scored
+    optional int32 goal_number = 21;
+
     // pseudo stats
     // is_turnover = !(goal || dribble || pass)
+
+    // DEPRECATED STATS
+    optional bool deprecated_field_goal_number = 11;
```
Comment (twobackfromtheend) [inline api/stats/events.proto]: do we want to highlight these more? (`DEPRECATED_goal_number`)

File: carball/json_parser/sanity_check/checks/player_data_frame_check.py [hunk 2]
```diff
@@ -6,12 +6,15 @@
...
 
+    excluded_columns = ['ball_cam', 'ping', 'double_jump_active']
+
     def critical_checks(self) -> List[Optional[AssertionError]]:
         player = self.obj
         if player.data is not None:
             return [
                 self.check(condition, f'player.data column: {column_name} has truthy values')
                 for column_name, condition in player.data.any(axis=0).iteritems()
+                if column_name not in self.excluded_columns
```
Comment (twobackfromtheend) [inline carball/json_parser/sanity_check/checks/player_data_frame_check.py]: lol when I wrote this I would've thought we'd do the opposite - have a list of stuff we care about. but this is good if it works.

File: carball/tests/analysis_test.py [hunk 3]
```diff
@@ -4,7 +4,9 @@
 import shutil
 import subprocess
 import traceback
+from io import BytesIO
 
+from carball.analysis.utils.pandas_manager import PandasManager
```
Comment (twobackfromtheend) [inline carball/tests/analysis_test.py]: I think you should ctrl-alt-o and ctrl-alt-L more often. will fix imports ordering and spacing.

File: carball/tests/analysis_test.py [hunk 4]
```diff
@@ -51,6 +53,8 @@ def __test_replays(BASE_DIR):
                                                        sanity_check=SanityChecker())
                 with open(os.path.join(OUTPUT_DIR, 'game.json'), 'w') as f:
                     f.write(MessageToJson(analysis_manager.protobuf_game))
+                pandas = PandasManager.safe_read_pandas_to_memory(BytesIO(analysis_manager.df_bytes))
+                print(len(pandas))
```
Comment (twobackfromtheend) [inline carball/tests/analysis_test.py]: why not logger

File: carball/tests/analysis_test.py [hunk 5]
```diff
@@ -51,6 +53,8 @@ def __test_replays(BASE_DIR):
                                                        sanity_check=SanityChecker())
                 with open(os.path.join(OUTPUT_DIR, 'game.json'), 'w') as f:
                     f.write(MessageToJson(analysis_manager.protobuf_game))
+                pandas = PandasManager.safe_read_pandas_to_memory(BytesIO(analysis_manager.df_bytes))
```
Comment (twobackfromtheend) [inline carball/tests/analysis_test.py]: I'd rather not use a library's name for a variable (though it obviously does not matter here)

File: api/stats/events.proto (hunk 1 above)
Comment (dtracers) [reply api/stats/events.proto]: thats a good idea

**Pull request merged**